dissect/target/plugins/apps/webservers/caddy.py

```diff
@@ -0,0 +1,74 @@
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Iterator
+
+from dissect.target import plugin
+from dissect.target.helpers.fsutil import open_decompress
+from dissect.target.plugins.apps.webservers.webservers import WebserverRecord
+from dissect.target.target import Target
+
+LOG_REGEX = re.compile(
+    r'(?P<remote_ip>.*?) - - \[(?P<ts>\d{2}\/[A-Za-z]{3}\/\d{4}:\d{2}:\d{2}:\d{2} (\+|\-)\d{4})\] "(?P<url>.*?)" (?P<status_code>\d{3}) (?P<bytes_sent>\d+)'  # noqa: E501
+)
+
+
+class CaddyPlugin(plugin.Plugin):
+    __namespace__ = "caddy"
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.log_paths = self.get_log_paths()
+
+    @plugin.internal
+    def get_log_paths(self) -> list[Path]:
+        log_paths = []
+        default_log_file = self.target.fs.path("/var/log/caddy_access.log")
+        if default_log_file.exists():
+            log_paths.append(default_log_file)
+
+        # Check for custom paths in Caddy config
+        if (config_file := self.target.fs.path("/etc/caddy/Caddyfile")).exists():
+            for line in config_file.open("rt"):
+                line = line.strip()
+                if "log  " in line:
+                    if (path := self.target.fs.path(line.split(" ")[1]).parent).exists():
+                        log_paths.append(path)
+                        break
+
+        return log_paths
+
+    def check_compatible(self) -> bool:
+        return self.target.fs.path(self.LOG_FILE_PATH).exists()
+
+    @plugin.export(record=WebserverRecord)
+    def access(self) -> Iterator[WebserverRecord]:
```

Comment:
Can you add a unit test for caddy log files?

Reply:
added in [760128a](https://github.com/fox-it/dissect.target/pull/127/commits/760128a6843d771e324b285adcfd3cc4368ea635)